sale_validity_auto_cancel/models/sale_order.py

```diff
@@ -28,6 +28,6 @@ def cron_sale_validity_auto_cancel(self):
             )
             for order in orders:
                 try:
-                    order.with_context(company_id=company.id)._action_cancel()
+                    order.with_company(company.id)._action_cancel()
```

Comment:
In fact, I'm not sure about the loop validity.

IMHO, you should search for sales with company of the current loop state.

Then, you should do `for order in orders.with_company(company.id):`

Reply:
True, should be fixed now